webhooks-extension/pkg/endpoints/webhook_test.go

```diff
@@ -67,1024 +103,632 @@ func TestGitHubSource(t *testing.T) {
+					expectedTriggers = append(expectedTriggers, t)
+				}
 			}
 		}
-	}
 
-	realOwnerAndRepo := ghSrc.Spec.OwnerAndRepository
-	if expectedOwnerAndRepo != realOwnerAndRepo {
-		t.Errorf("Incorrect OwnerAndRepository, expected %s but was: %s", expectedOwnerAndRepo, realOwnerAndRepo)
-	}
-	realGitHubAPIURL := ghSrc.Spec.GitHubAPIURL
-	if expectedGitHubAPIURL != realGitHubAPIURL {
-		t.Errorf("Incorrect GitHubAPIURL, expected %s but was: %s", expectedGitHubAPIURL, realGitHubAPIURL)
+		if !reflect.DeepEqual(el.Spec.Triggers, expectedTriggers) {
+			t.Errorf("Eventlistener trigger did not match expectation")
+			t.Errorf("got: %+v", el.Spec.Triggers)
+			t.Errorf("expected: %+v", expectedTriggers)
+		}
 	}
-}
 
-// Check the webhooks against the GET all webhooks
-func testGetAllWebhooks(expectedWebhooks []webhook, r *Resource, t *testing.T) {
-	httpReq := dummyHTTPRequest("GET", "http://wwww.dummy.com:8080/webhooks/", nil)
-	req := dummyRestfulRequest(httpReq, "")
-	httpWriter := httptest.NewRecorder()
-	resp := dummyRestfulResponse(httpWriter)
-	r.getAllWebhooks(req, resp)
-	actualWebhooks := []webhook{}
-	err := json.NewDecoder(httpWriter.Body).Decode(&actualWebhooks)
+	err = r.TriggersClient.TektonV1alpha1().EventListeners("install-namespace").Delete(el.Name, &metav1.DeleteOptions{})
 	if err != nil {
-		t.Errorf("Error decoding result into []webhook{}: %s", err.Error())
-		return
+		t.Error("Error occurred deleting eventlistener")
 	}
 
-	fmt.Printf("%+v", actualWebhooks)
+}
 
-	if len(expectedWebhooks) != len(actualWebhooks) {
-		t.Errorf("Incorrect length of result, expected %d, but was %d", len(expectedWebhooks), len(actualWebhooks))
-		return
+func TestDeleteFromEventListener(t *testing.T) {
+	var hooks = []webhook{
+		{
+			Name:             "name1",
+			Namespace:        installNs,
+			GitRepositoryURL: "https://github.com/owner/repo",
+			AccessTokenRef:   "token1",
+			Pipeline:         "pipeline1",
+			DockerRegistry:   "registry1",
+			HelmSecret:       "helmsecret1",
+			ReleaseName:      "releasename1",
+			PullTask:         "pulltask1",
+			OnSuccessComment: "onsuccesscomment1",
+			OnFailureComment: "onfailurecomment1",
+		},
+		{
+			Name:             "name2",
+			Namespace:        "foo",
+			GitRepositoryURL: "https://github.com/owner/repo",
+			AccessTokenRef:   "token2",
+			Pipeline:         "pipeline2",
+			DockerRegistry:   "registry2",
+			OnSuccessComment: "onsuccesscomment2",
+			OnFailureComment: "onfailurecomment2",
+		},
 	}
 
-	// Now compare the arrays expectedWebhooks and actualWebhooks by turning them into maps
-	expected := map[webhook]bool{}
-	actual := map[webhook]bool{}
-	for i := range expectedWebhooks {
-		if expectedWebhooks[i].DockerRegistry == "" {
-			expectedWebhooks[i].DockerRegistry = defaultRegistry
-		}
-		expected[expectedWebhooks[i]] = true
-		actual[actualWebhooks[i]] = true
+	r := dummyResource()
+	os.Setenv("SERVICE_ACCOUNT", "tekton-test-service-account")
```

Comment:
More service account handling

Reply:
In the test code the setting of a service account is generally done to ensure that the value is picked up and assigned correctly - not changing here, but will add service accounts to the list of items that might need extra test in the "extra tests needed" work item that needs opening.